adapters/repos/db/inverted/new_prop_length_tracker.go

```
@@ -0,0 +1,292 @@
+	t.Lock()
+	defer t.Unlock()
+
+	t.data.SumData[propName] = t.data.SumData[propName] - int(value)
+	t.data.CountData[propName] = t.data.CountData[propName] - 1
+
+	bucketId := t.bucketFromValue(value)
+	if _, ok := t.data.BucketedData[propName]; ok {
+		t.data.BucketedData[propName][int(bucketId)] = t.data.BucketedData[propName][int(bucketId)] - 1
+	} else {
+		return errors.New("property not found")
+	}
+
+	return nil
+}
+
+// Returns the bucket that the given value belongs to
+func (t *JsonPropertyLengthTracker) bucketFromValue(value float32) int {
+	if t.UnlimitedBuckets {
+		return int(value)
+	}
+	if value <= 5.00 {
+		return int(value) - 1
+	}
+
+	bucket := int(math.Log(float64(value)/4.0)/math.Log(1.25) + 4)
+	if bucket > 63 {
+		return 64
+	}
+	return int(bucket)
+}
+
+// Returns the value that the given bucket represents
+//
+//nolint:all
+func (t *JsonPropertyLengthTracker) valueFromBucket(bucket int) float32 {
+	if t.UnlimitedBuckets {
+		return float32(bucket)
+	}
+	if bucket <= 5 {
+		return float32(bucket + 1)
+	}
+
+	if bucket > 63 {
+		return -1
+	}
+
+	return float32(4 * math.Pow(1.25, float64(bucket)-3.5))
+}
+
+// Returns the average length of the given property
+func (t *JsonPropertyLengthTracker) PropertyMean(propName string) (float32, error) {
+	t.Lock()
+	defer t.Unlock()
+	sum, ok := t.data.SumData[propName]
+	if !ok {
+		return 0, nil
+	}
+	count, ok := t.data.CountData[propName]
+	if !ok {
+		return 0, nil
+	}
+
+	return float32(sum) / float32(count), nil
+}
+
+// returns totalPropertyLength, totalCount, average propertyLength = sum / totalCount, total propertylength, totalCount, error
+func (t *JsonPropertyLengthTracker) PropertyTally(propName string) (int, int, float64, error) {
+	t.Lock()
+	defer t.Unlock()
+	sum, ok := t.data.SumData[propName]
+	if !ok {
+		return 0, 0, 0, nil
+	}
+	count, ok := t.data.CountData[propName]
+	if !ok {
+		return 0, 0, 0, nil
```

also, should we return an error here?

If that doesn't make sense, can you remove the error from the return value? Then there is no codepath that returns an error

---

Backwards compatibility :(